app/org/sagebionetworks/bridge/dynamodb/DynamoExternalIdDao.java

```diff
@@ -140,22 +141,21 @@ public final void setMapper(DynamoDBMapper mapper) {
 
             if (list.getCount() > pageSize) {
                 // we retrieved more records from Dynamo than we are returning
-                offsetKey = identifiers.get(pageSize - 1).getIdentifier();
+                nextPageOffsetKey = identifiers.get(pageSize - 1).getIdentifier();
             } else {
                 // This is the last key, not the next key of the next page of records. It only exists if there's a record
                 // beyond the records we've converted to a page. Then get the last key in the list.
                 Map<String, AttributeValue> lastEvaluated = list.getLastEvaluatedKey();
-                offsetKey = lastEvaluated != null ? lastEvaluated.get(IDENTIFIER).getS() : null;
+                nextPageOffsetKey = lastEvaluated != null ? lastEvaluated.get(IDENTIFIER).getS() : null;
             }
-        } while ((identifiers.size() < pageSize) && (offsetKey != null));
-
-        ForwardCursorPagedResourceList<ExternalIdentifierInfo> resourceList = new ForwardCursorPagedResourceList<>(
-                identifiers, offsetKey, pageSize)
-                .withFilter(ID_FILTER, idFilter);
-        if (assignmentFilter != null) {
-            resourceList = resourceList.withFilter(ASSIGNMENT_FILTER, assignmentFilter.toString());
-        }
-        return resourceList;
+        } while ((identifiers.size() < pageSize) && (nextPageOffsetKey != null));
+
+        return new ForwardCursorPagedResourceList<>(
+                identifiers, nextPageOffsetKey)
+                .withRequestParam(ResourceList.OFFSET_KEY, offsetKey)
```

Comment:
Went through, found all the places where we're creating ForwardCursorPagedResourceLists, and ensured there are tests for 1) offsetKey/nextPageOffsetKey being set correctly; 2) all query parameters being returned in the map.